File: delivery_customization/custom_scripts/delivery_note.js
frappe.ui.form.on('Delivery Note', {
	setup: function(frm) {
		frm.set_query('company_shipping_address_c', function(doc) {
                        if(!doc.company) {
                                frappe.throw(__('Please set Company'));
                        }

                        return {
                                query: 'frappe.contacts.doctype.address.address.address_query',
                                filters: {
                                        link_doctype: 'Company',
                                        link_name: doc.company
                                }
                      };
	                });
	    },
	onload: function(frm) {
        frm.set_query("set_warehouse", function() {
    			return {
    				"filters": {
    					"company_address": frm.doc.company_address,
    					"is_group": 0
    				}
    			};
    		});

		frm.set_query("set_target_warehouse", function() {
			return {
				"filters": {
					"company_address": frm.doc.company_shipping_address_c,
					"is_group": 0
				}
			};
		});
	}
});

frappe.ui.form.on("Delivery Note", {
	refresh: function(frm) {
		if(frm.doc.set_warehouse && frm.doc.set_target_warehouse){
			frappe.call({
				"method":"delivery_customization.delivery_customization.delivery_note.check_warehouse_state",
				"freeze":true,
				"args":{
					"source_warehouse":frm.doc.set_warehouse,
					"target_warehouse":frm.doc.set_target_warehouse
				},
				"callback":function(res){
					console.log(res.message);
					if(res.message == 0){
						frm.remove_custom_button("E-Way Bill JSON", 'Create');
					}
				  
				}
			   });
		}
        
	        
	}
});

File: delivery_customization/delivery_customization/delivery_note.py
import frappe

@frappe.whitelist()
def check_warehouse_state(source_warehouse,target_warehouse):
    # frappe.msgprint(str(source_warehouse))
    # frappe.msgprint(str(target_warehouse))
    source_address = frappe.db.get_value("Warehouse",source_warehouse,"company_address")
    target_address = frappe.db.get_value("Warehouse",target_warehouse,"company_address")
    # frappe.msgprint(str(source_address))
    # frappe.msgprint(str(target_address))

    source_state = ""
    target_state = ""
    if source_warehouse:
        source_state = frappe.db.get_value("Address",source_address,"state")
        # frappe.msgprint(str(source_state))

    if target_warehouse:
        target_state = frappe.db.get_value("Address",target_address,"state")
        # frappe.msgprint(str(target_state))

    if source_state and target_state:
        if source_state == target_state:
            return 1
        else:
            return 0

